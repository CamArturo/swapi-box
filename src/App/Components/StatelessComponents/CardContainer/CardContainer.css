.card-container {
  margin: 4rem 0 7rem;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  overflow: hidden; }

/* Large Devices, Wide Screens */
/* Medium Devices, Desktops */
/* Small Devices, Tablets */
/* Extra Small Devices, Phones */
/* Custom, iPhone Retina */
