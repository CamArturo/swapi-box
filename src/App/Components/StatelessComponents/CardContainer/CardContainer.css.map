{
"version": 3,
"mappings": "AAAA,eAAgB;EACd,MAAM,EAAE,WAAW;EACnB,OAAO,EAAE,IAAI;EACb,SAAS,EAAE,IAAI;EACf,eAAe,EAAE,MAAM;EACvB,QAAQ,EAAE,MAAM;;AAGlB,iCAAiC;AAKjC,8BAA8B;AAK9B,4BAA4B;AAM5B,iCAAiC;AAKjC,2BAA2B",
"sources": ["CardContainer.scss"],
"names": [],
"file": "CardContainer.css"
}