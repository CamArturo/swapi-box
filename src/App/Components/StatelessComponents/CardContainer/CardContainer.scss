.card-container {
  margin: 5rem 0 7rem;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  //overflow: scroll;
  overflow: hidden;
}

/* Large Devices, Wide Screens */
@media only screen and (max-width: 1200px) {

}

/* Medium Devices, Desktops */
@media only screen and (max-width: 992px) {

}

/* Small Devices, Tablets */
@media only screen and (max-width: 768px) {

}


/* Extra Small Devices, Phones */
@media only screen and (max-width: 480px) {

}

/* Custom, iPhone Retina */
@media only screen and (max-width: 320px) {

}