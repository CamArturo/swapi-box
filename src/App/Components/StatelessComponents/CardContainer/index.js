import React from 'react';
import PropTypes from 'prop-types';

const CardContainer = props => {
  return (
    <div>

    </div>
  );
};

CardContainer.propTypes = {
  
};

export default CardContainer;
