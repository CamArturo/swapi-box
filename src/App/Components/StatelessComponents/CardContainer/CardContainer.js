import React from 'react';
import PropTypes from 'prop-types';
import Card from "../../StatefullComponents/Card/Card";
import './CardContainer.css';

const CardContainer =
  ({category, people, planets, vehicles, updateFavorites}) => {
    let array = [];
    if (category === 'people') {
      array = people;
    } else if (category === 'planets') {
      array = planets;
    } else {
      array = vehicles;
    }
    const allCards = array.map((card, index) => {
      return (
        <Card
          cardInfo={card}
          updateFavorites={updateFavorites}
          key={`card${index}`}
        />
      );
    });

    return (
      <div className="card-container">
        {allCards}
      </div>
    );
  };

CardContainer.propTypes = {
  category: PropTypes.string.isRequired,
  people: PropTypes.array.isRequired,
  planets: PropTypes.array.isRequired,
  vehicles: PropTypes.array.isRequired,
  updateFavorites: PropTypes.func.isRequired
};

export default CardContainer;
