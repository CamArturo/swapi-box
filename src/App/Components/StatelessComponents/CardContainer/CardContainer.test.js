import React from 'react';
import {shallow, mount} from 'enzyme';
import CardContainer from "./CardContainer";

describe('CardContainer test', () => {

  const category = 'people';
  const people = [
    {
      name: "Luke Skywalker",
      species: "Human",
      homeworld: "Tatooine",
      population: "200000",
      favorite: false
    },
    {
      name: "C-3PO",
      species: "Droid",
      homeworld: "Tatooine",
      population: "200000",
      favorite: false
    }
  ];

  const planets = [
    {
      name: "Yavin IV",
      terrain: "jungle, rainforests",
      population: "1000",
      climate: "temperate, tropical",
      residents: ""
    },
    {
      name: "Hoth",
      terrain: "tundra, ice caves, mountain ranges",
      population: "unknown",
      climate: "frozen",
      residents: ""
    }
  ];
  const vehicles = [
    {
      name: "Sand Crawler",
      model: "Digger Crawler",
      vehicleClass: "wheeled",
      passengers: "30",
      favorite: false
    },
    {
      name: "T-16 skyhopper",
      model: "T-16 skyhopper",
      vehicleClass: "repulsorcraft",
      passengers: "1",
      favorite: false
    }
  ];
  const favorites = [
    {
      name: "Sand Crawler",
      model: "Digger Crawler",
      vehicleClass: "wheeled",
      passengers: "30",
      favorite: false
    },
    {
      name: "T-16 skyhopper",
      model: "T-16 skyhopper",
      vehicleClass: "repulsorcraft",
      passengers: "1",
      favorite: false
    }
  ];

  const updateFavorites = jest.fn();
  let wrapper;
  beforeEach(() => {

    wrapper = shallow(<CardContainer category={category}
                                     people={people}
                                     planets={planets}
                                     vehicles={vehicles}
                                     favorites={favorites}
                                     updateFavorites={updateFavorites}
    />);
  });

  it('should match the snapshot', () => {

    // const wrapper = shallow(<CardContainer
    //
    // />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should assign array based on category passed in', () => {

    // const app = mount(<CardContainer category={category}
    //                                  people={people}
    //                                  planets={planets}
    //                                  vehicles={vehicles}
    //                                  favorites={favorites}
    //                                  updateFavorites={updateFavorites}
    // />);
    //
    // expect(app.find(Card).prop(category)).toBe('people');

    // expect(wrapper.find(card).prop).toBe(people);
  });
});