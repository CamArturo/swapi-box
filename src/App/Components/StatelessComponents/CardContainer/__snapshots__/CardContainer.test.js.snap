// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer test should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    category="people"
    favorites={
        Array [
            Object {
              "favorite": false,
              "model": "Digger Crawler",
              "name": "Sand Crawler",
              "passengers": "30",
              "vehicleClass": "wheeled",
            },
            Object {
              "favorite": false,
              "model": "T-16 skyhopper",
              "name": "T-16 skyhopper",
              "passengers": "1",
              "vehicleClass": "repulsorcraft",
            },
          ]
    }
    people={
        Array [
            Object {
              "favorite": false,
              "homeworld": "Tatooine",
              "name": "Luke Skywalker",
              "population": "200000",
              "species": "Human",
            },
            Object {
              "favorite": false,
              "homeworld": "Tatooine",
              "name": "C-3PO",
              "population": "200000",
              "species": "Droid",
            },
          ]
    }
    planets={
        Array [
            Object {
              "climate": "temperate, tropical",
              "name": "Yavin IV",
              "population": "1000",
              "residents": "",
              "terrain": "jungle, rainforests",
            },
            Object {
              "climate": "frozen",
              "name": "Hoth",
              "population": "unknown",
              "residents": "",
              "terrain": "tundra, ice caves, mountain ranges",
            },
          ]
    }
    updateFavorites={[Function]}
    vehicles={
        Array [
            Object {
              "favorite": false,
              "model": "Digger Crawler",
              "name": "Sand Crawler",
              "passengers": "30",
              "vehicleClass": "wheeled",
            },
            Object {
              "favorite": false,
              "model": "T-16 skyhopper",
              "name": "T-16 skyhopper",
              "passengers": "1",
              "vehicleClass": "repulsorcraft",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          cardInfo={
                    Object {
                              "favorite": false,
                              "homeworld": "Tatooine",
                              "name": "Luke Skywalker",
                              "population": "200000",
                              "species": "Human",
                            }
          }
          category="people"
          selectedClass="card"
          updateFavorites={[Function]}
/>,
        <Card
          cardInfo={
                    Object {
                              "favorite": false,
                              "homeworld": "Tatooine",
                              "name": "C-3PO",
                              "population": "200000",
                              "species": "Droid",
                            }
          }
          category="people"
          selectedClass="card"
          updateFavorites={[Function]}
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "card0",
        "nodeType": "function",
        "props": Object {
          "cardInfo": Object {
            "favorite": false,
            "homeworld": "Tatooine",
            "name": "Luke Skywalker",
            "population": "200000",
            "species": "Human",
          },
          "category": "people",
          "selectedClass": "card",
          "updateFavorites": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "card1",
        "nodeType": "function",
        "props": Object {
          "cardInfo": Object {
            "favorite": false,
            "homeworld": "Tatooine",
            "name": "C-3PO",
            "population": "200000",
            "species": "Droid",
          },
          "category": "people",
          "selectedClass": "card",
          "updateFavorites": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            cardInfo={
                        Object {
                                    "favorite": false,
                                    "homeworld": "Tatooine",
                                    "name": "Luke Skywalker",
                                    "population": "200000",
                                    "species": "Human",
                                  }
            }
            category="people"
            selectedClass="card"
            updateFavorites={[Function]}
/>,
          <Card
            cardInfo={
                        Object {
                                    "favorite": false,
                                    "homeworld": "Tatooine",
                                    "name": "C-3PO",
                                    "population": "200000",
                                    "species": "Droid",
                                  }
            }
            category="people"
            selectedClass="card"
            updateFavorites={[Function]}
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "card0",
          "nodeType": "function",
          "props": Object {
            "cardInfo": Object {
              "favorite": false,
              "homeworld": "Tatooine",
              "name": "Luke Skywalker",
              "population": "200000",
              "species": "Human",
            },
            "category": "people",
            "selectedClass": "card",
            "updateFavorites": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "card1",
          "nodeType": "function",
          "props": Object {
            "cardInfo": Object {
              "favorite": false,
              "homeworld": "Tatooine",
              "name": "C-3PO",
              "population": "200000",
              "species": "Droid",
            },
            "category": "people",
            "selectedClass": "card",
            "updateFavorites": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
